tests/basic/nomad_test.go

```diff
@@ -0,0 +1,415 @@
+		log.FailOnError(err, "Failed to get Nomad Client")
+	})
+
+	It("Should create a volume and validate it's active", func() {
+		volumeID := "test-volume"
+		pluginID := "portworx"
+		capacityMin := int64(1 * 1024 * 1024 * 1024)
+		capacityMax := capacityMin
+		accessMode := "single-node-writer"
+		attachmentMode := "file-system"
+		fsType := "ext4"
+		mountFlags := []string{}
+
+		err := client.CreateVolume(volumeID, pluginID, capacityMin, capacityMax, accessMode, attachmentMode, fsType, mountFlags)
+		log.FailOnError(err, "Failed to create volume")
+
+		// Validate the volume
+		volume, err := client.VolumeInfo(volumeID)
+		log.FailOnError(err, "Failed to get volume info")
+		volumes = append(volumes, volume)
+		if volume.ID == volumeID {
+			log.Infof("Volume %v created and validated successfully", volumeID)
+		} else {
+			log.Errorf(" Volume %v not created successfully")
+		}
+	})
+	JustAfterEach(func() {
+		for _, volume := range volumes {
+			err = client.DeleteVolume(volume.ID)
+			if err != nil {
+				log.Errorf("Volume %v could not be deleted with err %v", volume.ID, err)
+			}
+		}
+	})
+})
+
+var _ = Describe("{RunFioJobOnNomad}", func() {
+	var client *nomad.NomadClient
+	var err error
+	var volumeID string
+	var pluginID string
+
+	BeforeEach(func() {
+		StartTorpedoTest("RunFioJobOnNomad", "Runs FIO Job on Nomad Cluster", nil, 0)
+		client, err = nomad.NewNomadClient()
+		log.FailOnError(err, "Failed to get Nomad Client")
+		volumeID = "fio-test-volume"
+		pluginID = "portworx"
+		capacityMin := int64(20 * 1024 * 1024 * 1024)
+		capacityMax := capacityMin
+		accessMode := "single-node-writer"
+		attachmentMode := "file-system"
+		fsType := "ext4"
+		mountFlags := []string{}
+		err = client.CreateVolume(volumeID, pluginID, capacityMin, capacityMax, accessMode, attachmentMode, fsType, mountFlags)
+		log.FailOnError(err, "Failed to create volume")
+		log.Infof("Volume %v created successfully", volumeID)
+		jobID := "fio-job"
+		fioJob := client.CreateFioJobSpec(volumeID, jobID)
+		err = client.CreateJob(fioJob)
+		log.FailOnError(err, "Failed to create Fio job")
+		log.Infof("Fio job %v is created successfully", fioJob.ID)
+		log.Infof("Sleeping for 10 seconds to absorb startup time of allocation")
+		time.Sleep(10 * time.Second)
+	})
+	It("Run Fio job defined above and monitor its execution", func() {
+		const checkInterval = time.Minute * 1
+		const runDuration = time.Minute * 10
+		endTime := time.Now().Add(runDuration)
+		for time.Now().Before(endTime) {
+			status, err := client.CheckJobStatus("fio")
+			log.FailOnError(err, "Failed to check Fio job status")
+			if status != "running" {
+				log.Errorf("Fio job is not running")
+				return
+			}
+			log.Infof("Allocation and fio-task is running successfully as of now. Will check in 1 minute again.")
+			time.Sleep(checkInterval)
+		}
+		log.Infof("Fio ran successfully for the duration of the test")
+	})
+	AfterEach(func() {
+		err = client.DeleteJob("fio")
```

Comment:
seems this clean up is repeating. can we have cleanup function and call it everywhere

Reply:
Good catch - already changing this logic in my next PR that also involve more tests.